evalml/objectives/binary_classification_objective.py

```diff
@@ -1,15 +1,22 @@
 """Base class for all binary classification objectives."""
 import numpy as np
+import pandas as pd
 from scipy.optimize import differential_evolution
 
 from evalml.objectives.objective_base import ObjectiveBase
 from evalml.problem_types import ProblemTypes
+from evalml.utils.nullable_type_utils import _downcast_nullable_y
 
 
 class BinaryClassificationObjective(ObjectiveBase):
     """Base class for all binary classification objectives."""
 
     problem_types = [ProblemTypes.BINARY, ProblemTypes.TIME_SERIES_BINARY]
+
+    # Referring to the pandas nullable dtypes; not just woodwork logical types
+    _integer_nullable_incompatible = False
+    _boolean_nullable_incompatible = False
+
```

Comment:
Good question--0thanks for pointing it out! It's only binary classification objectives that have incompatibilities. Originally, I was going to put it in ObectiveBase, but  decided that it'd be better to go as specific as possible since the call to `_handle_nullable_types` will get called in `objective_function` (not `score` as I'd been planning at one point). 

I think my only holdup with having just BinaryClassificationObjectives have these properties is that then the others don't have any concept of nullable type compatibility. They don't really need to, which is why I decided to get as specific as possible, but I'm very open to changing it if you or someone else feels strongly.

Reply:
Makes sense. I'm inclined to go broader rather than more specific, for the reason you mentioned. It's not a guarantee that all of our non-binary objectives will remain compatible with nullable types, so it's safer to make it more broad. I don't think where `_handle_nullable_types` is called should have an impact, since we'd only need to call it whenever we have an incompatible objective.

Reply:
Moved nullable type logic to ObjectiveBase 6ddbf4745